crates/storage/provider/src/providers/static_file/manager.rs

```diff
@@ -82,14 +83,105 @@ impl StaticFileProvider {
     }
 
     /// Creates a new [`StaticFileProvider`] with read-only access.
-    pub fn read_only(path: impl AsRef<Path>) -> ProviderResult<Self> {
-        Self::new(path, StaticFileAccess::RO)
+    ///
+    /// `watch_directory` will keep track of changes on the directory and re-initialize its inner
+    /// index accordingly (eg. node pushes a new block).
+    pub fn read_only(path: impl AsRef<Path>, watch_directory: bool) -> ProviderResult<Self> {
```

Comment:
this can now panic if not within tokio runtime

I'd like to keep the function as is, and instead add `with_directory_watcher() -> Self`

Reply:
hmm, don't really like it, but I get your point, since we're only calling this a few times in the codebase this should be okay